src/basic/log.h

```diff
@@ -36,10 +36,11 @@ typedef enum LogTarget{
         _LOG_TARGET_INVALID = -1
 } LogTarget;
 
-#define LOG_REALM_PLUS_LEVEL(realm, level)      \
-        ((realm) << 10 | (level))
-#define LOG_REALM_REMOVE_LEVEL(realm_level)     \
-        ((realm_level >> 10))
+#define LOG_REALM_PLUS_LEVEL(realm, level)  ((realm) << 10 | (level))
+#define LOG_REALM_REMOVE_LEVEL(realm_level) ((realm_level) >> 10)
+#define SYNTHETIC_ERRNO(num)                (1u << 30 | (num))
+#define IS_SYNTHETIC_ERRNO(val)             (1u << 30 & (val))
```

Comment:
hmm, i think IS_XYZ() macros should probably return 0 or 1, i.e. maybe prefix this with `!!`? (or alternatively write as `((val) >> 30) & 1`?